src/groupByKey.ts

```diff
@@ -1,7 +1,13 @@
 type GroupsMap<T> = {
-  [key: string]: T[];
+  [key: string | number]: T[];
 };
 
-export function groupByKey(items, key) {
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
   // write code here;
+  return items.reduce((acc: GroupsMap<T>, item: T) => {
+    const value = item[key] as string | number;
+    const sortedItem = items.filter((el) => el[key] === value);
```

Comment:
The use of `items.filter((el) => el[key] === value)` is incorrect for grouping. This line filters the entire list for each item, which is inefficient and does not achieve the desired grouping. Instead, you should push the current `item` into the corresponding group in the accumulator `acc`.